registry/storage/manifeststore.go

```diff
@@ -130,5 +302,45 @@ func (ms *manifestStore) Delete(ctx context.Context, dgst digest.Digest) error {
 }
 
 func (ms *manifestStore) Enumerate(ctx context.Context, manifests []distribution.Manifest, last distribution.Manifest) (n int, err error) {
-	return 0, distribution.ErrUnsupported
+	context.GetLogger(ms.ctx).Debug("(*manifestStore).Enumerate")
+	dgsts, err := enumerateAllBlobs(ms.blobStore, ctx)
+	if err != nil && err != io.EOF {
+		return 0, err
+	}
+	found := make([]distribution.Manifest, 0, len(dgsts))
+	for _, dgst := range dgsts {
+		m, err := ms.Get(ctx, dgst)
+		if err == nil && ms.enumerateKind != manifestRevisionEnumOrphaned {
+			if cmpManifests(m, last) > 0 {
+				found = append(found, m)
+			}
+			continue
+		}
+
+		if err != nil && ms.enumerateKind != manifestRevisionEnumValid {
```

Comment:
This should probably distinguish between various err types and return different media types based on it.